src/protocol/sort/shuffle.rs

```diff
@@ -0,0 +1,113 @@
+use crate::helpers::prss::PrssSpace;
+use crate::helpers::Direction;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+use rand_chacha::ChaCha8Rng;
+
+#[allow(dead_code)]
+pub struct Shuffle {}
+
+impl Shuffle {
+    /// This implements Fisher Yates shuffle described here <https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle>
+    #[allow(dead_code)]
+    fn generate_random_permutation(
+        batchsize: usize,
+        direction: Direction,
+        index: u128,
+        prss: &PrssSpace,
+    ) -> Vec<usize> {
```

Comment:
It is a little concerning that `Permutation` crate requirements start to influence our code design decisions here - the fact that we return `Vec<usize>` is mainly driven by `Permutation::from_vec` reqs right? It would be great if we could address this soon

Reply:
Thats right. However, this function is returns a permutation in the range of usize. It doesn't have anything to do with Field. What do you think should be the intended behaviour here? @akoshelev